js/ChannelFactory.js

```diff
@@ -1,141 +1,150 @@
+        function Handle(channel, handle) {
+            this.unsubscribe = function () {
+                channel.unsubscribe(handle);
+            };
+        }
+
+        var setProps = function (myChannel, schema, scope, channelFactory, busName) {
+
+            myChannel.handle = uuid();
+
+            myChannel.schema = schema;
+
+            myChannel.channelName = schema.id;
+
+            myChannel.recreate = function (context) {
+                return channelFactory.get(schema, context, busName);
+            };
+
+            myChannel.publish = function (msg, callback) {
+                channelFactory.coreapi.publish(busName, myChannel.channelName, msg, callback, scope);
+            };
+
+            myChannel.unsubscribe = function (handle) {
+                //we expose this method publicly, but we only give access to the handle function itself
+                //so if the consumer uses this function directly, we need to re-call it via the handle in order to pass in the actual uuid
+                if (typeof handle === "object") {
+                    handle.unsubscribe();
+                } else {
+                    channelFactory.coreapi.unsubscribe(busName, myChannel.channelName, scope, handle);
+                }
+            };
+
+            myChannel.subscribe = function (callback, filterPredicate, captureSelfPublished) {
+                var handle = channelFactory.coreapi.subscribe(busName, myChannel.channelName, callback, scope, filterPredicate, captureSelfPublished);
+                return new Handle(myChannel, handle);
+            };
+
+            myChannel.subscribeOnce = function (callback, filterPredicate, captureSelfPublished) {
+                var handle = channelFactory.coreapi.subscribe(busName, myChannel.channelName, callback, scope, filterPredicate, captureSelfPublished, true);
+                return new Handle(myChannel, handle);
+            };
+        },
+        //setup the resolvers with (1) a default that returns schema objects directly (first check),
+        //and (2) any configured resolver that does other lookup such as by name
+        resolvers = [function (schema) {
+            if (typeof schema === "object") {
+                return schema;
+            }
+        }];
+
+        if (config.resolver) {
+            resolvers.push(config.resolver);
+        }
+
+        /**
+         * Given a name of a schema iterate over the resolver array until you find a match.
+         * @param name
+         * @return {*}
+         */
+        this.resolveSchema = function (name) {
+                   
+            var schema;
+            resolvers.some(function (res) {
+                schema = res(name);
+                return schema;
+            }, this);
+            
+          //The case we don't want to lookup and utilize a schema
+            //*
+            if (schema === undefined) {
+                return { "id": "default/DefaultChannel" };

```

Comment:
This id should use the `name` param passed to the resolveSchema function, so that any number of channels can be created instead of all having the same name of "default/DefaultChannel".